deploy/runtools/run_farm_deploy_managers.py

```diff
@@ -148,6 +148,13 @@ def disconnect_all_nbds_instance(self) -> None:
 
                 run("; ".join(fullcmd))
 
+    def get_remote_sim_dir_for_slot(self, slotno: int) -> str:
+        """ Returns the path on the remote for a given slot number. This function
+        must return a path that includes the trailing slash."""
+        remote_home_dir = self.parent_node.get_sim_dir()
+        remote_sim_dir = """{}/sim_slot_{}/""".format(remote_home_dir, slotno)
```

Comment:
Rather than stating the assumption about the return value in the docstring, it makes sense to put in an assertion here.  Also, use formatstrings instead of `.format()`
```suggestion
        """ Returns the path on the remote for a given slot number. """
        
        remote_home_dir = self.parent_node.get_sim_dir()
        remote_sim_dir = f"{remote_home_dir}/sim_slot_{slotno}/"
        assert remote_sim_dir[-1] == '/', 
            f"Return value of get_remote_sim_dir_for_slot({remote_home_dir}, {slotno}) must end with '/'."
        # so that callers can supply the return value to rsync and have well-defined semantics
```

NOTE: I'm not sure that I have legal indenting in my suggestion and I'm just taking a guess at why the trailing slash matters but it would be good to document why we need it, possibly in a comment if the assertion message becomes too unweildly.